python/src/aac/lang/definition_helpers.py

```diff
@@ -0,0 +1,266 @@
+
+    Args:
+        parsed_definitions: The list of parsed definitions to search.
+        root_key: The root key to filter on.
+
+    Returns:
+        A list of ParedDefinitions with the given root key AaC model definitions.
+    """
+
+    def does_definition_root_match(parsed_definition: ParsedDefinition) -> bool:
+        return root_key == parsed_definition.get_root_key()
+
+    return list(filter(does_definition_root_match, parsed_definitions))
+
+
+def get_definition_by_name(definition_name: str, parsed_definitions: list[ParsedDefinition]) -> Optional[ParsedDefinition]:
+    """Get a definition of models of a specific root name.
+
+    The aac.parser.parse() returns a dict of all parsed types.  Sometimes it is
+    useful to only work with certain roots (i.e. model or data).  This utility
+    method allows a setup of parsed models to be "filtered" to a specific root name.
+
+    Args:
+        models: The dict of models to search.
+        definition_name: The definition's name to locate.
+
+    Returns:
+        A ParsedDefinition with the given name or None.
+    """
+
+    def does_definition_name_match(parsed_definition: ParsedDefinition) -> bool:
+        return remove_list_type_indicator(definition_name) == parsed_definition.name
+
+    matching_definitions = list(filter(does_definition_name_match, parsed_definitions))
+    matching_definitions_count = len(matching_definitions)
+
+    result = None
+    if matching_definitions_count > 1:
+        logging.error('Found multiple definitions with the same name "{definition_name}"\n{matching_definitions}"')
+        raise RuntimeError(f'Found multiple definitions with the same name "{definition_name}"\n{matching_definitions}')
+
+    elif matching_definitions_count < 1:
+        logging.debug('Failed to find a definition with the name "{definition_name}"')
+
+    else:
+        result = matching_definitions[0]
+
+    return result
+
+
+def convert_parsed_definitions_to_dict_definition(parsed_definitions: list[ParsedDefinition]) -> dict:
+    """
+    DEPRECATED: Returns a combined dict from the list of ParsedDefinitions.
+
+    This function is intended only as a stop-gap to support some implementations until they have been fully
+    converted to utilize ParsedDefinitions.
+    """
+    return {definition.name: definition.definition for definition in parsed_definitions}
+
+
+def remove_list_type_indicator(list_type: str) -> str:
+    """Return the AaC type without any trailing list indicator characters '[]'."""
+    return list_type.split("[]")[0]
+
+
+def is_array_type(type: str) -> bool:
+    """Returns a boolean indicating if the field is an array of multiple entries."""
+    return type.endswith("[]")
+
+
+def get_definition_fields_and_types(
+    parsed_definition: ParsedDefinition, context_definitions: list[ParsedDefinition]
+) -> dict[str, ParsedDefinition]:
+    """Return a list of field names to their definition types."""
```

Comment:
So, I tried to avoid using the term "mapped" since python is out here trying to be all fancy with dictionaries instead of maps. I didn't want to inject my java bias, but if you also agree that "mapped" is a good description for dictionaries, then I'm happy to make the change.

Reply:
I don't think "mapped" implies a Java bias and I think most devs of any experience would understand the different terminology for the data structure, in Common Lisp we call them hash tables :shrug: - it's pretty much all the same thing regardless.

Reply:
Agreed - I'll accept the suggestion.